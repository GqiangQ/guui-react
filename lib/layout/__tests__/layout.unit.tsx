import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Layout from '../layout';

configure({adapter: new Adapter()});

describe(Layout.displayName, () => {
  it('DisplayName', () => {
    expect(Layout.displayName).toEqual('Layout');
  });
  describe('props: dir', () => {
    it('default = vertical', () => {
      const tree = renderer
        .create(<Layout></Layout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('= horizontal', () => {
      const tree = renderer
        .create(<Layout dir="horizontal"></Layout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('props: className', () => {
    it('= xxx', () => {
      const tree = renderer
        .create(<Layout className="xxx"></Layout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('props: style', () => {
    it(`= {color: 'red'}`, () => {
      const tree = renderer
        .create(<Layout style={{color: 'red'}}></Layout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
