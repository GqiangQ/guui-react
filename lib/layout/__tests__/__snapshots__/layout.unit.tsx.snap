// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout props: className = xxx 1`] = `
<div
  className="gu-layout gu-layout-vertical xxx"
/>
`;

exports[`Layout props: dir = horizontal 1`] = `
<div
  className="gu-layout gu-layout-horizontal "
/>
`;

exports[`Layout props: dir default = vertical 1`] = `
<div
  className="gu-layout gu-layout-vertical "
/>
`;

exports[`Layout props: style = {color: 'red'} 1`] = `
<div
  className="gu-layout gu-layout-vertical "
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
